lib/logstash/outputs/google_bigquery.rb

```diff
@@ -296,20 +295,62 @@ def initialize_temp_directory
   end
 
   ##
+  # If logstash fails for any reason, this method will attempt to recover any
+  # files that were in flight during the crash. The recovery works in two
+  # phases. All files with .bqjob file entries will be put into the
+  # delete_queue. All files without .bqjob files will be put into the
+  # upload_queue.
+  #
+  # If a temp_directory configuration is defined, this method will look in the
+  # defined directory. If temp_directory is not defined, this method will look
+  # in all /tmp/logstash-bq-* directories.
+  #
+  # Note that there is still a small race condition where a file might have
+  # been uploaded, but the .bqjob file wasn't saved yet. In such a case, the
+  # file will be loaded into BigQuery twice.
+  def recover_temp_directories
+    if @temp_directory.empty?
+      Dir.entries('/tmp').select{
+        |entry| File.directory? File.join('/tmp',entry) and entry.start_with?('logstash-bq-')
+      }.each {
+        |entry| recover_temp_directory(File.join('/tmp',entry))
+      }
+    else
+      recover_temp_directory(@temp_directory)
+    end
+  end
+
+  ##
+  # List all files in a directory, and add them to either the upload_queue or
+  # delete_queue.
+  def recover_temp_directory(directory)
+    # Recover files that have been uploaded, but not deleted.
+    Dir.glob(File.join(directory, "*.bqjob")).each do |bqjob_filename|
+      job_id = File.open(bqjob_filename, 'r') {|f| f.read}.strip
+      filename = File.join(directory, File.basename(bqjob_filename, ".bqjob"))
+      @logger.info("BQ: resuming job.",
```

Comment:
I am fine either way. I thought it was worth making these verbose since it doesn't happen often, and it's good to know when files are recovered (even when not in debug mode). I'll leave it to you--would you like me to change it?